### PR TITLE
Add DC TANF program

### PR DESCRIPTION
**Pull request by Chiukai-C**
Thanks for contributing! Please remove any top-level sections that do not apply to your changes.

- [ ] `make format && make documentation` has been run.

# New variable

- [ ] Label field added
- [ ] Documentation field added
- [ ] Unit field added
- [ ] Default value field added if relevant
- [ ] Variable name follows conventions
- [ ] Unit test(s) added
- [ ] Integration test(s) added if relevant
- [ ] Issues this PR fixes linked

## What's changed

Description of the changes here.

# Bug fix

- [ ] Regression test added
- [ ] Regression test passing

## What this fixes and how it's fixed

Description of how this fix works goes here. Link any issues this PR fixes.

### CODE REVIEW

**Comment by Chiukai-C** [comment]
Add DC TANF tests, parameters, variables and restructure the code to make it consistent

**Comment by MaxGhenis** [comment]
Try:
* `pip install -U black`
* `make format`
* Commit
* `git pull upstream master`

**Comment by Chiukai-C** [comment]
Still not working. I try make format several times before. Maybe something wrong with the PR version?

File: policyengine_us/variables/gov/states/dc/dhs/tanf/income/dc_tanf_countable_gross_unearned_income.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    unit = USD
+    definition_period = YEAR
+    defined_for = StateCode.DC
+
+    def formula(spm_unit, period, parameters):
+        p = parameters(period).gov.states.dc.dhs.tanf.income
+        # Sum unearned sources, plus child support disregard $150 received per month.
+        gross_unearned = add(spm_unit, period, p.unearned)
+        child_support = add(spm_unit, period, ["child_support_received"])
+        return gross_unearned + max_(child_support - 150 * 12, 0)
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/dc/dhs/tanf/income/dc_tanf_countable_gross_unearned_income.py:17]: Please parameterize the `150`

Also use `MONTHS_IN_YEAR` rather than 12

File: policyengine_us/variables/gov/states/dc/dhs/tanf/eligibility/dc_tanf_resources_eligible.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        p = parameters(period).gov.states.dc.dhs.tanf.resources_limit
+        person = spm_unit.members
+        # Check if the household has at least one elderly member.
+        has_elderly = spm_unit.any(
+            person("age", period) >= p.elderly_age_threshold
+        )
+        # Check if the household has at least one disabled member.
+        has_disabled = spm_unit.any(person("is_disabled", period))
+        # Look up resource limit by the condition.
+        resource_limit = p.main[has_elderly or has_disabled]
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/dc/dhs/tanf/eligibility/dc_tanf_resources_eligible.py:22]: For vectorization
```suggestion
        resource_limit = p.main[has_elderly | has_disabled]
```

File: policyengine_us/parameters/gov/states/dc/dhs/tanf/resources_limit/main.yaml
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/dc/dhs/tanf/resources_limit/main.yaml:1]: Ideally we could do this but breakdowns by bools doesn't show up well in the app. Could you please split these into two files?

Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/dc/dhs/tanf/resources_limit/main.yaml:1]: Could you rename the folder to `resource_limit`?

File: policyengine_us/variables/gov/states/dc/dhs/tanf/eligibility/dc_tanf_eligible.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    entity = SPMUnit
+    label = "DC TANF eligible"
+    definition_period = YEAR
+    defined_for = StateCode.DC
+
+    def formula(spm_unit, period, parameters):
+        demographic_eligible = spm_unit("is_demographic_tanf_eligible", period)
+        income_eligible = spm_unit("dc_tanf_income_eligible", period)
+        resources_eligible = spm_unit("dc_tanf_resources_eligible", period)
+        return demographic_eligible and income_eligible and resources_eligible
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/dc/dhs/tanf/eligibility/dc_tanf_eligible.py:15]: Needed for vectorization
```suggestion
        return demographic_eligible & income_eligible & resources_eligible
```

**Review by MaxGhenis — COMMENTED** [review]
Looks great, just a few changes

File: policyengine_us/parameters/gov/states/dc/dhs/tanf/resources_limit/main.yaml
Comment (Chiukai-C) [reply policyengine_us/parameters/gov/states/dc/dhs/tanf/resources_limit/main.yaml:1]: Like what they did in New York TANF? Sure I can do that if that's what you want.

Comment (MaxGhenis) [reply policyengine_us/parameters/gov/states/dc/dhs/tanf/resources_limit/main.yaml:1]: yep

File: policyengine_us/variables/gov/states/dc/dhs/tanf/eligibility/dc_tanf_eligible.py (hunk 3 above)
Comment (Chiukai-C) [reply policyengine_us/variables/gov/states/dc/dhs/tanf/eligibility/dc_tanf_eligible.py:15]: If I use '&' instead of 'and', things will go wrong when I make test.

Comment (MaxGhenis) [reply policyengine_us/variables/gov/states/dc/dhs/tanf/eligibility/dc_tanf_eligible.py:15]: What's the error?

Comment (Chiukai-C) [reply policyengine_us/variables/gov/states/dc/dhs/tanf/eligibility/dc_tanf_eligible.py]: I could not make test now. I got this.
`coverage run -a --branch -m policyengine_core.scripts.policyengine_command test policyengine_us/tests/policy/ -c policyengine_us
process_begin: CreateProcess(NULL, coverage run -a --branch -m policyengine_core.scripts.policyengine_command test policyengine_us/tests/policy/ -c policyengine_us, ...) failed.
make (e=2): The system cannot find the file specified.
make: *** [Makefile:9: test] Error 2`

Comment (Chiukai-C) [reply policyengine_us/variables/gov/states/dc/dhs/tanf/eligibility/dc_tanf_eligible.py]: If I change 'and' back to '&'. I got this.
`policyengine_us/variables/gov/states/dc/dhs/tanf/eligibility/dc_tanf_eligible.py:15: in formula
    return demographic_eligible & income_eligible & resources_eligible
E   TypeError: ufunc 'bitwise_and' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''`
I searched on Internet and they said it is due to operator precedence. Maybe I try `return (demographic_eligible) & (income_eligible) & (resources_eligible)`

Comment (Chiukai-C) [reply policyengine_us/variables/gov/states/dc/dhs/tanf/eligibility/dc_tanf_eligible.py]: I figure it out! I set float value type instead of bool for resouces_eligible. That's why it got error.

File: policyengine_us/variables/gov/states/dc/dhs/tanf/dc_tanf_grant_standard.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
+from policyengine_us.model_api import *
+
+
+class dc_tanf_grant_standard(Variable):
+    value_type = float
+    entity = SPMUnit
+    label = "DC TANF grand standard"
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/dc/dhs/tanf/dc_tanf_grant_standard.py:7]:
```suggestion
    label = "DC TANF grant standard"
```

File: policyengine_us/variables/gov/states/dc/dhs/tanf/income/dc_tanf_countable_gross_unearned_income.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    unit = USD
+    definition_period = YEAR
+    defined_for = StateCode.DC
+
+    def formula(spm_unit, period, parameters):
+        p = parameters(period).gov.states.dc.dhs.tanf.income
+        # Sum unearned sources, plus child support disregard $150 received per month.
+        gross_unearned = add(spm_unit, period, p.unearned)
+        child_support = add(spm_unit, period, ["child_support_received"])
+        return gross_unearned + max_(child_support - 150 * MONTHS_IN_YEAR, 0)
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/dc/dhs/tanf/income/dc_tanf_countable_gross_unearned_income.py:17]: Make $150 a parameter

File: policyengine_us/variables/gov/states/dc/dhs/tanf/income/dc_tanf_countable_income_for_computing_benefits.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    value_type = float
+    entity = SPMUnit
+    label = "DC TANF countable earned income"
+    unit = USD
+    definition_period = YEAR
+    defined_for = StateCode.DC
+
+    def formula(spm_unit, period, parameters):
+        gross_earnings = spm_unit("dc_tanf_gross_earned_income", period)
+        gross_unearnings = spm_unit(
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/dc/dhs/tanf/income/dc_tanf_countable_income_for_computing_benefits.py:14]: Unearnings deserves to be a word but it sadly is not
```suggestion
        gross_unearned_income = spm_unit(
```

File: policyengine_us/variables/gov/states/dc/dhs/tanf/income/dc_tanf_countable_income_for_computing_benefits.py [hunk 7]
```diff
@@ -0,0 +1,23 @@
+from policyengine_us.model_api import *
+
+
+class dc_tanf_countable_income_for_computing_benefits(Variable):
+    value_type = float
+    entity = SPMUnit
+    label = "DC TANF countable earned income"
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/dc/dhs/tanf/income/dc_tanf_countable_income_for_computing_benefits.py:7]:
```suggestion
    label = "DC TANF countable income"
```

File: policyengine_us/variables/gov/states/dc/dhs/tanf/income/dc_tanf_countable_income_for_computing_benefits.py [hunk 8]
```diff
@@ -0,0 +1,23 @@
+from policyengine_us.model_api import *
+
+
+class dc_tanf_countable_income_for_computing_benefits(Variable):
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/dc/dhs/tanf/income/dc_tanf_countable_income_for_computing_benefits.py:4]: Could we shorten this to `dc_tanf_countable_income`?

File: policyengine_us/variables/gov/states/dc/dhs/tanf/income/dc_tanf_countable_income_for_computing_benefits.py [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+    defined_for = StateCode.DC
+
+    def formula(spm_unit, period, parameters):
+        gross_earnings = spm_unit("dc_tanf_gross_earned_income", period)
+        gross_unearnings = spm_unit(
+            "dc_tanf_countable_gross_unearned_income", period
+        )
+        p = parameters(period).gov.states.dc.dhs.tanf.income.earned_deduction
+        annual_flat_exclusion = p.flat * MONTHS_IN_YEAR
+        return (
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/dc/dhs/tanf/income/dc_tanf_countable_income_for_computing_benefits.py:19]: add tests

File: policyengine_us/variables/gov/states/dc/dhs/tanf/income/dc_tanf_countable_gross_unearned_income.py [hunk 10]
```diff
@@ -14,4 +14,9 @@ def formula(spm_unit, period, parameters):
         # Sum unearned sources, plus child support disregard $150 received per month.
         gross_unearned = add(spm_unit, period, p.unearned)
         child_support = add(spm_unit, period, ["child_support_received"])
-        return gross_unearned + max_(child_support - 150 * MONTHS_IN_YEAR, 0)
+        monthly_child_support = add(
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/dc/dhs/tanf/income/dc_tanf_countable_gross_unearned_income.py]:
```suggestion
        child_support_deduction = MONTHS_IN_YEAR * add(
```

**Review by MaxGhenis — COMMENTED** [review]
also make format

File: policyengine_us/parameters/gov/states/dc/dhs/tanf/need_standard/main.yaml [hunk 11]
```diff
@@ -9,6 +9,7 @@ metadata:
   reference:
     - title: TANF for District Families
       href: https://dhs.dc.gov/service/tanf-district-families
+    # TODO: Add reference to legal code, and go up to whatever it says.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/dc/dhs/tanf/need_standard/main.yaml:12]: please do the todo or remove it and move to an issue

File: policyengine_us/variables/gov/states/dc/dhs/tanf/income/dc_tanf_countable_earned_income.py [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+
+    def formula(spm_unit, period, parameters):
+        gross_earnings = spm_unit("dc_tanf_gross_earned_income", period)
+        p = parameters(period).gov.states.dc.dhs.tanf.income.earned_deduction
+        enrolled = spm_unit("is_tanf_enrolled", period)
+        annual_flat_exclusion = p.flat * MONTHS_IN_YEAR
+        return where(
+            enrolled,
+            # For enrolled recipients, DC applies a flat and a percent deduction.
+            max_(gross_earnings - annual_flat_exclusion, 0)
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/dc/dhs/tanf/income/dc_tanf_countable_earned_income.py:20]: Since this is used twice can you make it a python variable, e.g. `earnings_after_flat_exclusion`

File: policyengine_us/variables/gov/states/dc/dhs/tanf/income/dc_tanf_countable_income.py [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+    defined_for = StateCode.DC
+
+    def formula(spm_unit, period, parameters):
+        gross_earnings = spm_unit("dc_tanf_gross_earned_income", period)
+        gross_unearned_income = spm_unit(
+            "dc_tanf_countable_gross_unearned_income", period
+        )
+        p = parameters(period).gov.states.dc.dhs.tanf.income.earned_deduction
+        annual_flat_exclusion = p.flat * MONTHS_IN_YEAR
+        return (
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/dc/dhs/tanf/income/dc_tanf_countable_income.py:19]: To make the code a bit more self-documenting
```suggestion
        earnings_after_flat_exclusion = max_(gross_earnings - annual_flat_exclusion, 0)
        countable_earned_income = earnings_after_flat_exclusion * (1 - p.percentage)
        return countable_earned_income + gross_unearned_income
```
